sklearn/ensemble/bagging.py

```diff
@@ -838,13 +874,13 @@ def _validate_estimator(self):
         super(BaggingRegressor, self)._validate_estimator(
             default=DecisionTreeRegressor())
 
-    def _set_oob_score(self, X, y):
+    def _set_oob_score(self, X, y, n_more_estimators):
         n_samples = y.shape[0]
 
         predictions = np.zeros((n_samples,))
         n_predictions = np.zeros((n_samples,))
 
-        for estimator, samples, features in zip(self.estimators_,
+        for estimator, samples, features in zip(self.estimators_[-n_more_estimators:],
```

Comment:
`estimators_samples_` and `estimators_features_` are for the last used batch of samples in `fit()` and `estimators_` contains all estimators trained so far. So you have to select only the last `n_more_estimators`